pkg/hubble/parser/threefour/parser.go

```diff
@@ -87,11 +97,16 @@ func New(
 		linkGetter:     linkGetter,
 		epResolver:     common.NewEndpointResolver(log, endpointGetter, identityGetter, ipGetter),
 		packet:         packet,
+		enabled:        args.EnableL3L4PolicyEnrichment,
 	}, nil
 }
 
 // Decode decodes the data from 'data' into 'decoded'
 func (p *Parser) Decode(data []byte, decoded *pb.Flow) error {
+	if !p.enabled {
```

Comment:
I don't believe we should skip all enrichment for layer 3 and 4, only the policy correlation happening in L235.

Reply:
+1